File: backend/member-service/src/main/java/com/example/memberservice/config/FeignConfig.java
package com.example.memberservice.config;

import com.example.memberservice.global.decoder.FeignErrorDecoder;
import feign.Logger;
import feign.Retryer;
import feign.codec.ErrorDecoder;
import org.springframework.context.annotation.Bean;

public class FeignConfig {

    @Bean
    Logger.Level feignLoggerLevel() {
        return Logger.Level.FULL;
    }

    @Bean
    public Retryer retryer() {
        return new Retryer.Default();
    }

    @Bean
    public ErrorDecoder errorDecoder() {
        return new FeignErrorDecoder();
    }
}

File: backend/member-service/src/main/kotlin/com/example/memberservice/config/FeignConfig.kt
package com.example.memberservice.config

import com.example.memberservice.global.decoder.FeignErrorDecoder
import feign.Logger
import feign.Retryer
import feign.codec.ErrorDecoder
import org.springframework.context.annotation.Bean

class FeignConfig {

    @Bean
    fun feignLoggerLevel(): Logger.Level = Logger.Level.FULL

    @Bean
    fun retryer(): Retryer = Retryer.Default()

    @Bean
    fun errorDecoder(): ErrorDecoder = FeignErrorDecoder()
}
